File: HumanEvalJs/40.js
//// BEGIN - PROMPT
/**
 * triplesSumToZero takes an array of integers as an input.
 * It returns true if there are three distinct elements in the array that
 * sum to zero, and false otherwise.
 *
 * Examples:
 *   triplesSumToZero([1, 3, 5, 0]); // false
 *   triplesSumToZero([1, 3, -2, 1]); // true
 *   triplesSumToZero([1, 2, 3, 7]); // false
 *   triplesSumToZero([2, 4, -5, 3, 9, 7]); // true
 *   triplesSumToZero([1]); // false
 */

function triplesSumToZero(l) {
//// BEGIN - CANONICAL SOLUTION
    for (let i = 0; i < l.length; i++) {
        for (let j = i + 1; j < l.length; j++) {
            for (let k = j + 1; k < l.length; k++) {
                if (l[i] + l[j] + l[k] === 0) {
                    return true;
                }
            }
        }
    }
    return false;
}

//// BEGIN - TEST
const METADATA = {};

const assert = require('assert');

function check(candidate) {
    assert.strictEqual(candidate([1, 3, 5, 0]), false);
    assert.strictEqual(candidate([1, 3, 5, -1]), false);
    assert.strictEqual(candidate([1, 3, -2, 1]), true);
    assert.strictEqual(candidate([1, 2, 3, 7]), false);
    assert.strictEqual(candidate([1, 2, 5, 7]), false);
    assert.strictEqual(candidate([2, 4, -5, 3, 9, 7]), true);
    assert.strictEqual(candidate([1]), false);
    assert.strictEqual(candidate([1, 3, 5, -100]), false);
    assert.strictEqual(candidate([100, 3, 5, -100]), false);
}

//// BEGIN - CHECK
check(triplesSumToZero);

File: HumanEval/40.py
## BEGIN - PROMPT


def triples_sum_to_zero(l: list):
    """
    triples_sum_to_zero takes a list of integers as an input.
    it returns True if there are three distinct elements in the list that
    sum to zero, and False otherwise.

    >>> triples_sum_to_zero([1, 3, 5, 0])
    False
    >>> triples_sum_to_zero([1, 3, -2, 1])
    True
    >>> triples_sum_to_zero([1, 2, 3, 7])
    False
    >>> triples_sum_to_zero([2, 4, -5, 3, 9, 7])
    True
    >>> triples_sum_to_zero([1])
    False
    """

## BEGIN - CANONICAL SOLUTION
    for i in range(len(l)):
        for j in range(i + 1, len(l)):
            for k in range(j + 1, len(l)):
                if l[i] + l[j] + l[k] == 0:
                    return True
    return False

## BEGIN - TEST


METADATA = {}


def check(candidate):
    assert candidate([1, 3, 5, 0]) == False
    assert candidate([1, 3, 5, -1]) == False
    assert candidate([1, 3, -2, 1]) == True
    assert candidate([1, 2, 3, 7]) == False
    assert candidate([1, 2, 5, 7]) == False
    assert candidate([2, 4, -5, 3, 9, 7]) == True
    assert candidate([1]) == False
    assert candidate([1, 3, 5, -100]) == False
    assert candidate([100, 3, 5, -100]) == False


## BEGIN - CHECK
check(triples_sum_to_zero)